apps/src/blockly/addons/cdoXml.js

```diff
@@ -35,8 +35,39 @@ export default function initializeBlocklyXml(blocklyWrapper) {
   };
 
   blocklyWrapper.Xml.domToBlockSpace = function(blockSpace, xml) {
+    // To position the blocks, we first render them all to the Block Space
+    //  and parse any X or Y coordinates set in the XML. Then, we store
+    //  the rendered blocks and the coordinates in an array so that we can
+    //  position them in two passes.
+    //  In the first pass, we position the visible blocks. In the second
+    //  pass, we position the invisible blocks. We do this so that
+    //  invisible blocks don't cause the visible blocks to flow
+    //  differently, which could leave gaps between the visible blocks.
+    const blocks = [];
+    /**
+     * NodeList.forEach() is not supported on IE. Use Array.prototype.forEach.call() as a workaround.
+     * https://developer.mozilla.org/en-US/docs/Web/API/NodeList/forEach
+     */
+    Array.prototype.forEach.call(xml.childNodes, function(xmlChild) {
+      if (xmlChild.nodeName.toLowerCase() !== 'block') {
+        // skip non-block xml elements
+        return;
+      }
+      const blockly_block = Blockly.Xml.domToBlock(xmlChild, blockSpace);
+      const x = parseInt(xmlChild.getAttribute('x'), 10);
+      const y = parseInt(xmlChild.getAttribute('y'), 10);
+      blocks.push({
+        blockly_block: blockly_block,
+        x: x,
+        y: y
+      });
+    });
+
     const metrics = blockSpace.getMetrics();
-    const width = metrics ? metrics.viewWidth : 0;
+    const viewWidth = metrics ? metrics.viewWidth : 0;
+    const width = blocks[0]
```

Comment:
Nit: lot of widths being thrown around here, is this `blockWidth`?

Reply:
I swear I had this as `blockWidth` at one point time! Good call though - fixed.